packages/test-schemas/src/media.ts

```diff
@@ -1,36 +1,120 @@
+  item: Node! @relationDocument(property: "itemID")
 }
 
-interface VisualMedia @createModel(accountRelation: LIST, description: "An interface for visual media objects") {
+interface Collection @createModel(description: "An interface for a collection of documents") {
+  name: String! @string(maxLength: 50)
+  description: String @string(maxLength: 2000)
+  items: [CollectionItem] @relationFrom(model: "CollectionItem", property: "collectionID")
+  itemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "collectionID")
+}
+
+interface VisualMedia @createModel(description: "An interface for visual media objects") {
   width: Int! @int(min: 1)
   height: Int! @int(min: 1)
 }
 
-interface TimeMedia @createModel(accountRelation: LIST, description: "An interface for time media objects") {
-  duration: Int! @int(min: 0)
+interface TimeMedia @createModel(description: "An interface for time media objects") {
+  duration: Duration!
 }
 
-type ImageMedia implements MediaMetadata & VisualMedia @createModel(accountRelation: LIST, description: "A model for image objects") {
+interface ImageMedia implements MediaMetadata & VisualMedia @createModel(description: "An interface for image objects") {
   src: String! @string(maxLength: 500)
   size: Int
   width: Int! @int(min: 1)
   height: Int! @int(min: 1)
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
 }
 
-type AudioMedia implements MediaMetadata & TimeMedia @createModel(accountRelation: LIST, description: "A model for audio objects") {
+interface AudioMedia implements MediaMetadata & TimeMedia @createModel(description: "An interface for audio objects") {
   src: String! @string(maxLength: 500)
   size: Int
-  duration: Int! @int(min: 0)
+  duration: Duration!
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
 }
 
-type VideoMedia implements MediaMetadata & VisualMedia & TimeMedia @createModel(accountRelation: LIST, description: "A model for image objects") {
+interface VideoMedia implements MediaMetadata & VisualMedia & TimeMedia @createModel(description: "An interface for video objects") {
   src: String! @string(maxLength: 500)
   size: Int
-  duration: Int! @int(min: 0)
+  duration: Duration!
   width: Int! @int(min: 1)
   height: Int! @int(min: 1)
+  collectionItems: [CollectionItem] @relationFrom(model: "CollectionItem", property: "itemID")
+  collectionItemsCount: Int! @relationCountFrom(model: "CollectionItem", property: "itemID")
+}
+
+type MyImage implements ImageMedia
+  @createModel(description: "A model for image objects")
+  @createIndex(fields: [{ path: ["src"] }])
+  @createIndex(fields: [{ path: ["width"] }])
+  @createIndex(fields: [{ path: ["height"] }]) {
+  src: String! @string(maxLength: 500)
```

Comment:
can i have extra custom fields in here too, or no?

Reply:
Yes! The models implementing interfaces can add extra fields as needed